plugins/core/propagation_test.go

```diff
@@ -0,0 +1,240 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package core
+
+import (
+	"reflect"
+	"testing"
+)
+
+const testValueString = "test-value"
+
+type fields struct {
+	TraceID               string
+	ParentSegmentID       string
+	ParentService         string
+	ParentServiceInstance string
+	ParentEndpoint        string
+	AddressUsedAtClient   string
+	ParentSpanID          int32
+	Sample                int8
+}
+
+type args struct {
+	header string
+}
+
+func TestSpanContext_DecodeSW8(t *testing.T) {
+	tests := []struct {
+		name    string
+		fields  *fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name:   "Empty Header",
+			fields: nil,
+			args: args{
+				header: "",
+			},
+			wantErr: true,
+		},
+		{
+			name:   "Insufficient Header Entities",
+			fields: nil,
+			args: args{
+				header: "1-MWYyZDRiZjQ3YmY3MTFlYWI3OTRhY2RlNDgwMDExMjI=-MWU3YzIwNGE3YmY3MTFlYWI4NThhY2RlNDgwMDExMjI=",
+			},
+			wantErr: true,
+		},
+		{
+			name: "normal",
+			fields: &fields{
+				Sample:                1,
+				TraceID:               "1f2d4bf47bf711eab794acde48001122",
+				ParentSegmentID:       "1e7c204a7bf711eab858acde48001122",
+				ParentSpanID:          0,
+				ParentService:         "service",
+				ParentServiceInstance: "instance",
+				ParentEndpoint:        "propagation",
+				AddressUsedAtClient:   "propagation:5566",
+			},
+			args: args{
+				header: "1-MWYyZDRiZjQ3YmY3MTFlYWI3OTRhY2RlNDgwMDExMjI=-MWU3YzIwNGE3YmY3MTFlYWI4NThhY2RlNDgwMDExMjI=" +
+					"-0-c2VydmljZQ==-aW5zdGFuY2U=-cHJvcGFnYXRpb24=-cHJvcGFnYXRpb246NTU2Ng==",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tc := &SpanContext{}
+			if err := tc.DecodeSW8(tt.args.header); (err != nil) != tt.wantErr {
+				t.Errorf("DecodeSW8() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.fields != nil {
+				if tc.Sample != tt.fields.Sample {
```

Comment:
You can save a lot of time and do better assertions by using something like `assert`, see https://pkg.go.dev/github.com/stretchr/testify/assert

Also in this case you just fail but don't tell why so it is hard to debug.